docs/adrs/0000-caching-dependencies.md

```diff
@@ -0,0 +1,59 @@
+# 0. Caching dependencies
+Date: 2021-05-21
+
+Status: Proposed
+
+# Context
+`actions/setup-node` is the 2nd most popular action in GitHub Actions. A lot of customers use it in conjunction with [actions/cache](https://github.com/actions/cache) to speed up dependencies installation.  
+See more examples on proper usage in [actions/cache documentation](https://github.com/actions/cache/blob/main/examples.md#node---npm).
+
+# Goals & Anti-Goals
+Integration of caching functionality into `actions/setup-node` action will bring the following benefits for action users:
+- Decrease the entry threshold for using the cache for Node.js dependencies and simplify initial configuration
+- Simplify YAML pipelines because no need additional steps to enable caching
+
+As a result, more users will use the cache for Node.js builds and will be happy with fast builds.  
```

Comment:
I would move speed up to the benefits list, because it can save cost of the workflow